### PR TITLE
Handle transient S3 errors in tune

### PR DESCRIPTION
**Pull request by EthanMarx**
The remote tuning was being plagued by transient permission errors when fetching / pushing to s3. After chatting with nautilus admins, it was discovered that there is an issue with the internal url used for accessing s3 from within the cluster. The external url (although slower) works fine, so for the time being we use that. 

This PR also adds retries to various points where we interact with s3 to future proof us from this issue.

### CODE REVIEW

File: projects/data/environment.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@ dependencies:
   - lalsuite
   - astropy
   - bilby>1.4,<2
+  - pycbc
```
Comment (wbenoit26) [inline projects/data/environment.yaml:21]: What is this used for?

Comment (EthanMarx) [reply projects/data/environment.yaml:21]: I was working on `pycbc` wf gen in another branch, let me remove it

**Comment by wbenoit26** [comment]
Is the `conda-lock` accurate? Did you re-lock after removing pycbc?

**Comment by EthanMarx** [comment]
Yeah I relocked it - I can double check pycbc isn't in there

**Review by wbenoit26 — APPROVED** [review]
That's fine, I was just surprised to still see so many changes to it

**Comment by EthanMarx** [comment]
I think it's probably just versions of existing packages being updated but I can double check before merging

**Pull request merged**